### PR TITLE
add tests for calculate function

### PR DESCRIPTION
**Pull request by BenOrf**
- added a test for the calculate function
- tests if the calculator works correct

### CODE REVIEW

**Review by mbosselmann — COMMENTED** [review]
Looks good! 👍 Like your tests! :)

File: src/components/Base.js [hunk 1]
```diff
@@ -2,15 +2,15 @@ import Article from "../components/article/Article";
...
-    });
-    return price.toFixed(2);
-  }
+export const calculateSum = (articles) => {
+  let price = 0;
+  articles.forEach((article) => {
+    price += parseFloat(article.price.replace(",", ".")) * article.amount;
+  });
+  return price.toFixed(2);
+};
```
Comment (mpagels) [inline src/components/Base.js:11]: You're passing an argument here, but in line 25 in your `Base` component, you are calling it without a parameter. Is this working?

Comment (mpagels) [inline src/components/Base.js:11]: I highly recommend extracting this function here and put in a `service` folder and import the function from there back here.

File: src/components/Base.js [hunk 2]
```diff
@@ -2,15 +2,15 @@ import Article from "../components/article/Article";
...
 
-function Base({ setArticles, articles }) {
-  function calculateSum() {
-    let price = 0;
-    articles.forEach((article) => {
-      price += parseFloat(article.price.replace(",", ".")) * article.amount;
-    });
-    return price.toFixed(2);
-  }
+export const calculateSum = (articles) => {
```
Comment (mpagels) [inline src/components/Base.js:5]: Not a fan of the name of the function. IMHO, it's too generic. I would not assume that I have to pass an array with an articles object in it. I would call it `calculateSumOfArticles`

File: src/components/Base.js (hunk 1 above)
Comment (mpagels) [inline src/components/Base.js:11]: When you have time for a refactor, you can replace the `forEach` method with the `reduce` method. The `reduce` method was "designed" for "add/subtract" things.

Read more about reduce [here](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/Reduce).

File: src/components/base.test.js [hunk 3]
```diff
@@ -0,0 +1,55 @@
+import { getByRole, render, fireEvent } from "@testing-library/react";
+import { screen } from "@testing-library/dom";
+import userEvent from "@testing-library/user-event";
```
Comment (mpagels) [inline src/components/base.test.js:3]: Why do you import these? You don't use it at all.

File: src/components/base.test.js [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+      amount: 7,
+    },
+    {
+      price: "0,49",
+      amount: 4,
+    },
+  ];
+  const productSum = calculateSum(articles);
+  expect(productSum).toEqual("378.67");
+});
```
Comment (mpagels) [inline src/components/base.test.js:55]: Would add a test, that tests, what happens when you don't pass a parameter to the `calculateSum` function. Or you pass a parameter that is not an `array`. Or the `array` aren't holding objects with the correct properties.

**Review by mpagels — COMMENTED** [review]
Hey Ben, I love that you implemented tests for the function. I found some improvements, please have a look. Good work by the way.

**Pull request merged**